cedar-policy-validator/src/entity_manifest.rs

```diff
@@ -0,0 +1,1103 @@
+                data: (),
+            }
+        }
+
+        EntityRecordKind::Entity(_) | EntityRecordKind::AnyEntity => {
+            // no need to load data for entities, which are compared
+            // using ids
+            AccessTrie::new()
+        }
+    }
+}
+
+/// Given an expression, get the corresponding data path
+/// starting with a variable.
+/// If the expression is not a `<datapath-expr>`, return an error.
+/// See [`FailedAnalysisError`] for more information.
+fn get_expr_path(
+    expr: &Expr<Option<Type>>,
+    policy_id: &PolicyID,
+) -> Result<AccessPath, EntityManifestError> {
+    Ok(match expr.expr_kind() {
+        ExprKind::Slot(slot_id) => {
+            if slot_id.is_principal() {
+                AccessPath {
+                    root: EntityRoot::Var(Var::Principal),
+                    path: vec![],
+                    parents_required: false,
+                }
+            } else {
+                assert!(slot_id.is_resource());
+                AccessPath {
+                    root: EntityRoot::Var(Var::Resource),
+                    path: vec![],
+                    parents_required: false,
+                }
+            }
+        }
+        ExprKind::Var(var) => AccessPath {
+            root: EntityRoot::Var(*var),
+            path: vec![],
+            parents_required: false,
+        },
+        ExprKind::GetAttr { expr, attr } => {
+            let mut slice = get_expr_path(expr, policy_id)?;
+            slice.path.push(attr.clone());
+            slice
+        }
+        ExprKind::Lit(Literal::EntityUID(literal)) => AccessPath {
+            root: EntityRoot::Literal((**literal).clone()),
+            path: vec![],
+            parents_required: false,
+        },
+        ExprKind::Unknown(_) => Err(EntityManifestError::PartialExpressionError)?,
+        // all other variants of expressions result in failure to analyze.
+        _ => Err(EntityManifestError::FailedAnalysis(FailedAnalysisError {
+            source_loc: expr.source_loc().cloned(),
+            policy_id: policy_id.clone(),
+            expr_kind: expr.expr_kind().clone(),
+        }))?,
+    })
+}
+
+#[cfg(test)]
+mod entity_slice_tests {
+    use cedar_policy_core::{ast::PolicyID, extensions::Extensions, parser::parse_policy};
+
+    use super::*;
+
+    #[test]
+    fn test_simple_entity_manifest() {
+        let mut pset = PolicySet::new();
+        let policy = parse_policy(
+            None,
+            "permit(principal, action, resource)
+when {
+    principal.name == \"John\"
+};",
+        )
+        .expect("should succeed");
+        pset.add(policy.into()).expect("should succeed");
+
+        let schema = ValidatorSchema::from_str_natural(
+            "
+entity User = {
+  name: String,
+};
+
+entity Document;
+
+action Read appliesTo {
+  principal: [User],
+  resource: [Document]
+};
+        ",
+            Extensions::all_available(),
+        )
+        .unwrap()
+        .0;
```

Comment:
Since this schema is used repeatedly, we can just make a `schema()` helper function in this module that returns it